### ISSUE TITLE
RPG-Openworld: Write more intricate tests for edges_new, edges_init, and edges_free

### ISSUE THREAD

**Issue by rohandeme** [uchicago-cs/chiventure#1511]
We recently completed a pull request, where we added functions called edges_new, edges_init, and edges_free, which would allow us to add and free a 2D adjacency matrix of edges whose values represent the likelihood of two rooms generating next to one another. In this issue, we will add more detailed tests to ensure that these functions are working as expected.

**Comment by williamhu314** [comment]
Specifically, we will add tests with more matrices involved, including edge cases such as when wanting to make a 1x1 matrix or when wanting to make a bigger matrix (e.g. 20x20).

**Comment by rohandeme** [comment]
By the end of this issue, we intend to add tests for 1x1, 2x2, 3x3, and 4x4 matrices. Will add more updates when we come across edges cases that may be important to test as well.

**Comment by rohandeme** [comment]
We finished writing tests for edges_init and edges_new, which have been pushed to the openworld/edges branch.

**Comment by rohandeme** [comment]
We finished writing tests for edges_free, thus completing this issue. While we ran into no serious bugs or edge cases, writing more tests was quite valuable, as it allowed us to discover subtleties in the code that could be accounted for by future openworld teams. For example, there can be functions that validate that the matrix is a square (same number of rows as columns), and another that makes sure that the edge values are indeed between 0 and 5. We will make sure to include these recommendations in our final report.
